internal/util/helpers_test.go

```diff
@@ -4,18 +4,765 @@
+			grpcPort:      8090,
+			allowInsecure: true,
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			envToken:      "MINDER_AUTH_TOKEN",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to localhost without passing MINDER_AUTH_TOKEN as an argument",
+			grpcHost:      "127.0.0.1",
+			grpcPort:      8090,
+			allowInsecure: false,
+			issuerUrl:     "http://localhost:8081",
+			clientId:      "minder-cli",
+			envToken:      "",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to remote server with secure connection",
+			grpcHost:      "api.stacklok.com",
+			grpcPort:      443,
+			allowInsecure: false,
+			issuerUrl:     "https://auth.stacklok.com",
+			clientId:      "minder-cli",
+			envToken:      "MINDER_AUTH_TOKEN",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to remote server with insecure connection",
+			grpcHost:      "api.stacklok.com",
+			grpcPort:      443,
+			allowInsecure: true,
+			issuerUrl:     "https://auth.stacklok.com",
+			clientId:      "minder-cli",
+			envToken:      "MINDER_AUTH_TOKEN",
+			expectedError: false,
+		},
+		{
+			name:          "Valid GRPC connection to remote server without passing MINDER_AUTH_TOKEN as an argument",
+			grpcHost:      "api.stacklok.com",
+			grpcPort:      443,
+			allowInsecure: true,
+			issuerUrl:     "https://auth.stacklok.com",
+			clientId:      "minder-cli",
+			envToken:      "",
+			expectedError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			originalEnvToken := os.Getenv(util.MinderAuthTokenEnvVar)
+			err := os.Setenv(util.MinderAuthTokenEnvVar, tt.envToken)
+			if err != nil {
+				t.Errorf("error setting %v: %v", util.MinderAuthTokenEnvVar, err)
+			}
+			// reset this the environment variable when complete.
+			defer os.Setenv(util.MinderAuthTokenEnvVar, originalEnvToken)
+
+			conn, err := util.GetGrpcConnection(tt.grpcHost, tt.grpcPort, tt.allowInsecure, tt.issuerUrl, tt.clientId)
+			if (err != nil) != tt.expectedError {
+				t.Errorf("expected error: %v, got: %v", tt.expectedError, err)
+			}
+			if conn != nil {
+				err = conn.Close()
+				if err != nil {
+					t.Errorf("error closing connection: %v", err)
+				}
+			}
+
+		})
+	}
+}
+
+// TestSaveCredentials tests the SaveCredentials function
+func TestSaveCredentials(t *testing.T) {
+	t.Parallel()
+
+	tokens := util.OpenIdCredentials{
+		AccessToken:          "test_access_token",
+		RefreshToken:         "test_refresh_token",
+		AccessTokenExpiresAt: time.Time{}.Add(7 * 24 * time.Hour),
+	}
+
+	// Create a temporary directory
+	testDir := t.TempDir()
+
+	SetEnvVar(t, "XDG_CONFIG_HOME", testDir)
+
+	cfgPath := filepath.Join(testDir, "minder")
+
+	expectedFilePath := filepath.Join(cfgPath, "credentials.json")
+
+	filePath, err := util.SaveCredentials(tokens)
+	if err != nil {
+		t.Fatalf("expected no error, got %v", err)
+	}
```

Comment:
Slight preference for `require.NoError(t, err)` here and below, to break up the common-case flow of the test a bit less.

Reply:
will do that.